src/core/meshcop/joiner.cpp

```diff
@@ -296,6 +312,14 @@ void Joiner::HandleJoinerEntrust(Coap::Header &aHeader, Message &aMessage, const
 
     otLogInfoMeshCoP("join success!\r\n");
 
+    // configure a random Extended Address
+    for (size_t i = 0; i < sizeof(extAddress); i++)
+    {
+        extAddress.m8[i] = static_cast<uint8_t>(otPlatRandomGet());
+    }
+
+    mNetif.GetMac().SetExtAddress(extAddress);
```

Comment:
Need to update link local address here:
`mNetif.GetMle().UpdateLinkLocalAddress();`